src/Components/SavedSearchAlert/Components/SavedSearchAlertNameInput.tsx

```diff
@@ -0,0 +1,103 @@
+import { Input } from "@artsy/palette"
+import { SavedSearchAlertFormValues } from "Components/SavedSearchAlert/types"
+import { SystemQueryRenderer } from "System/Relay/SystemQueryRenderer"
+import { useFormikContext } from "formik"
+import { FC, useEffect, useMemo, useState } from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+import { SavedSearchAlertNameInputQuery } from "__generated__/SavedSearchAlertNameInputQuery.graphql"
+import { useSavedSearchAlertContext } from "Components/SavedSearchAlert/SavedSearchAlertContext"
+import { debounce } from "lodash"
+import { SavedSearchAlertNameInput_previewSavedSearch$data } from "__generated__/SavedSearchAlertNameInput_previewSavedSearch.graphql"
+import { useFeatureFlag } from "System/useFeatureFlag"
+
+export const SavedSearchAlertNameInputQueryRenderer: FC = () => {
+  const { criteria } = useSavedSearchAlertContext()
+  const [criteriaState, setCriteriaState] = useState({})
+
+  const debouncedSetCriteriaState = useMemo(
+    () => debounce(setCriteriaState, 200),
+    []
+  )
+
+  useEffect(() => {
+    debouncedSetCriteriaState(criteria)
+  }, [criteria, debouncedSetCriteriaState])
+
+  return (
+    <SystemQueryRenderer<SavedSearchAlertNameInputQuery>
+      query={graphql`
+        query SavedSearchAlertNameInputQuery(
+          $attributes: PreviewSavedSearchAttributes!
+        ) {
+          viewer {
+            previewSavedSearch(attributes: $attributes) {
+              ...SavedSearchAlertNameInput_previewSavedSearch
+            }
+          }
+        }
+      `}
+      variables={{ attributes: criteriaState }}
+      render={({ props, error }) => {
+        if (error) {
+          console.error(error)
+        }
+
+        return (
+          <SavedSearchAlertNameInputFragmentContainer
+            // @ts-expect-error "No overload matches this call" TODO: why?
+            previewSavedSearch={props?.viewer?.previewSavedSearch}
+          />
```

Comment:
Another issue I wasn't able to resolve yet. Seemed like a straightforward refactor to use this fragment container, but perhaps I missed something.